polars/src/chunked_array/kernels/take.rs

```diff
@@ -68,61 +68,44 @@ pub(crate) fn take_no_null_utf8(arr: &StringArray, indices: &UInt32Array) -> Arc
         .expect("out of memory");
     let offset_typed = offset_buf.typed_data_mut();
 
-    // The required size is yet unknown
-    // Allocate 1.5 times the expected size.
-    // where expected size is the length of bytes multiplied by the factor (take_len / current_len)
-    let mut values_capacity =
-        ((arr.value_data().len() as f32 * 1.5) as usize) / arr.len() * indices.len() as usize;
-    let mut values_buf = MutableBuffer::new(values_capacity);
-
     let mut length_so_far = 0;
     offset_typed[0] = length_so_far;
+    let mut values_buf;
 
     let nulls;
 
+    // the two loops are ~30% faster then a single loop in benchmarks.
+    offset_typed
+        .iter_mut()
+        .skip(1)
+        .enumerate()
+        .for_each(|(idx, offset)| {
```

Comment:
FYI, I couldn't replicate the better performance with two loops, but maybe there are slight differences between Arrow on master and polars (and Arrow 2.0). Also using a mutable buffer turned out to be slower than creating / copying a Vec for the values.

Final PR is here, which is still ~20% faster https://github.com/apache/arrow/pull/8814

Reply:
Nice work!

Curious that doing the `memcopy` is cheaper than directly writing to a Buffer. 
The Bechmark I ran had relatively large strings. So I guess there can be some differences with initial startup cost of a Buffer related to Vec? Another reason to root for a `Vec` buffer. :rocket: